ReactiveCocoa/Swift/Signal.swift

```diff
@@ -263,36 +263,43 @@ private func observeWithStates<T, U, E>(signalState: CombineLatestState<T>, othe
 	}, interrupted: onInterrupted)
 }
 
-/// Combines the latest value of the receiver with the latest value from
-/// the given signal.
-///
-/// The returned signal will not send a value until both inputs have sent
-/// at least one value each. If either signal is interrupted, the returned signal
-/// will also be interrupted.
-public func combineLatestWith<T, U, E>(otherSignal: Signal<U, E>)(signal: Signal<T, E>) -> Signal<(T, U), E> {
+private func combineLatestWith<T, U, V, E>(otherSignal: Signal<U, E>, transform: ((T, U) -> V))(signal: Signal<T, E>) -> Signal<V, E> {
 	return Signal { observer in
 		let lock = NSRecursiveLock()
 		lock.name = "org.reactivecocoa.ReactiveCocoa.combineLatestWith"
-
+		
 		let signalState = CombineLatestState<T>()
 		let otherState = CombineLatestState<U>()
-
+		
 		let onBothNext = { () -> () in
-			let combined = (signalState.latestValue!, otherState.latestValue!)
-			sendNext(observer, combined)
+			sendNext(observer, transform(signalState.latestValue!, otherState.latestValue!))
 		}
-
+		
 		let onError = { sendError(observer, $0) }
 		let onBothCompleted = { sendCompleted(observer) }
 		let onInterrupted = { sendInterrupted(observer) }
-
+		
 		let signalDisposable = signal |> observeWithStates(signalState, otherState, lock, onBothNext, onError, onBothCompleted, onInterrupted)
 		let otherDisposable = otherSignal |> observeWithStates(otherState, signalState, lock, onBothNext, onError, onBothCompleted, onInterrupted)
-
+		
 		return CompositeDisposable(ignoreNil([ signalDisposable, otherDisposable ]))
 	}
 }
 
+/// Combines the latest value of the receiver with the latest value from
+/// the given signal.
+///
+/// The returned signal will not send a value until both inputs have sent
+/// at least one value each. If either signal is interrupted, the returned signal
+/// will also be interrupted.
+public func combineLatestWith<T, U, E>(otherSignal: Signal<U, E>)(signal: Signal<T, E>) -> Signal<(T, U), E> {
+	return signal |> combineLatestWith(otherSignal) { ($0, $1) }
+}
+
+public func combineLatestWith<T, E>(otherSignal: Signal<T, E>)(signal: Signal<[T], E>) -> Signal<[T], E> {
```

Comment:
I don't think we'd want this variation to be public. A single `combineLatest` that supports sequences should be good enough.

If users want this specific operator, they can approximate it through `combineLatestWith` and `map`.